### PR TITLE
WIP: Polymorphic serialization fixed.

### PR DESCRIPTION
**Pull request by Northburns**
Don't merge until #110 resolved 🙂

### CODE REVIEW

File: src/test/kotlin/io/github/kryszak/gwatlin/util/ResourcesUtil.kt [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package io.github.kryszak.gwatlin.util
+
+object ResourcesUtil {
```
Comment (Kryszak) [inline src/test/kotlin/io/github/kryszak/gwatlin/util/ResourcesUtil.kt:3]: Let's make this object internal just for codacy complains

File: src/test/kotlin/io/github/kryszak/gwatlin/http/serializers/SerialNameDelegateTest.kt [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+import io.github.kryszak.gwatlin.api.gamemechanics.model.facts.BuffConversion
+import io.github.kryszak.gwatlin.api.guild.model.log.GuildLogInvited
+import io.github.kryszak.gwatlin.api.items.model.item.ArmorItem
+import io.github.kryszak.gwatlin.api.items.model.item.ItemRarity
+import io.github.kryszak.gwatlin.api.items.model.item.ItemType
+import io.kotest.core.spec.style.FunSpec
+import io.kotest.matchers.shouldBe
+
+class SerialNameDelegateTest : FunSpec({
```
Comment (Kryszak) [inline src/test/kotlin/io/github/kryszak/gwatlin/http/serializers/SerialNameDelegateTest.kt:11]: internal modifier is missing
+ 
extend ShouldSpec to keep consistent pattern in tests

**Review by Kryszak — CHANGES_REQUESTED** [review]
Fine improvement 👌

File: src/test/kotlin/io/github/kryszak/gwatlin/util/ResourcesUtil.kt (hunk 1 above)
Comment (Northburns) [reply src/test/kotlin/io/github/kryszak/gwatlin/util/ResourcesUtil.kt]: Fixed :+1:

File: src/test/kotlin/io/github/kryszak/gwatlin/http/serializers/SerialNameDelegateTest.kt (hunk 2 above)
Comment (Northburns) [reply src/test/kotlin/io/github/kryszak/gwatlin/http/serializers/SerialNameDelegateTest.kt]: Fixed :+1:

**Review by Kryszak — APPROVED** [review]

**Pull request merged**